app/src/main/java/de/westnordost/streetcomplete/quests/crossing/AddCrossing.kt

```diff
@@ -0,0 +1,186 @@
+package de.westnordost.streetcomplete.quests.crossing
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Node
+import de.westnordost.streetcomplete.data.osm.mapdata.Way
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment
+import de.westnordost.streetcomplete.util.initialBearingTo
+import de.westnordost.streetcomplete.util.normalizeDegrees
+
+class AddCrossing : OsmElementQuestType<Boolean> {
+
+    private val roadsFilter by lazy { """
+        ways with
+          highway ~ primary|primary_link|secondary|secondary_link|tertiary|tertiary_link|unclassified|residential
+          and area != yes
+          and crossing != no
+          and (access !~ private|no or (foot and foot !~ private|no))
+    """.toElementFilterExpression() }
+
+    private val footwaysFilter by lazy { """
+        ways with
+          (highway ~ footway|steps or highway ~ path|cycleway and foot ~ designated|yes)
+          and footway != sidewalk
+          and area != yes
+          and access !~ private|no
+    """.toElementFilterExpression() }
+
+    override val commitMessage = "Add whether there is a crossing"
+    override val wikiLink = "Tag:highway=crossing"
+    override val icon = R.drawable.ic_quest_pedestrian
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_crossing_title
+
+    override fun isApplicableTo(element: Element): Boolean? =
+        if(element !is Node || element.tags.isNotEmpty()) false else null
+
+    override fun getApplicableElements(mapData: MapDataWithGeometry): Iterable<Element> {
+
+        val roadsByNodeId = mapData.ways.asSequence()
+            .filter { roadsFilter.matches(it) }
+            .groupByNodeIds()
+        /* filter out nodes of roads that are the end of a road network (dead end), f.e.
+         * https://www.openstreetmap.org/node/280046349 or
+         * https://www.openstreetmap.org/node/56606744 */
+        roadsByNodeId.removeEndNodes()
+
+        /* require all roads at a shared node to either have no sidewalk tagging or all of them to
+         * have sidewalk tagging: If the sidewalk tagging changes at that point, it may be an
+         * indicator that this is the transition point between separate sidewalk mapping and
+         * sidewalk mapping on road-way. F.e.:
+         * https://www.openstreetmap.org/node/1839120490 */
+        val anySidewalk = setOf("both","left","right")
```

Comment:
is sidewalk=separate skipped deliberately?

With `sidewalk=separate` and footway crossing road this quest still makes sense. See https://www.openstreetmap.org/#map=19/50.06687/19.92738

![screen06](https://user-images.githubusercontent.com/899988/128859390-baa1d8eb-01e8-4943-a99a-036644c06962.png)

tagging crossing there would enable say tactile paving quest

Reply:
I think you misunderstood that portion of the code. See the explanatory comment above.

But yeah, separate footways are skipped, because of the discussion I had with @jyasskin : If there is indeed no crossing on separately mapped sidewalks, setting `crossing=no` is not enough, but the whole crossing way needs to be deleted which is something that StreetComplete can't do.